realm/kotlin-extensions/src/androidTest/kotlin/io/realm/rule/BlockingLooperThread.kt

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.rule
+
+import android.os.Handler
+import android.os.Looper
+import io.realm.TestHelper
+import io.realm.TestHelper.LooperTest
+import io.realm.internal.android.AndroidCapabilities
+import org.junit.runners.model.MultipleFailureException
+import java.io.Closeable
+import java.io.PrintStream
+import java.util.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.collections.ArrayList
+
+/**
+ * Helper class that makes it easier to run a piece of code inside a Looper Thread. This is done by
+ * calling the `run` method. This method will block until the Looper either completes or throws
+ * an exception.
+ *
+ * Usage:
+ * ```
+ * val lopperThread = LooperThreadTest()
+ *
+ * @Before
+ * fun setUp() {
+ *     // Runs before test
+ * }
+ *
+ * @After
+ * fun tearDown() {
+ *     // Runs after test completed or failed
+ * }
+ *
+ * @Test
+ * fun myTest() = looperThread.run {
+ *     // test code
+ * }
+ * ```
+ */
+class BlockingLooperThread {
```

Comment:
We should avoid copying files between `realm-library` and `kotlin-extensions`. In the past we used a shared sourceset to keep these files in. Doesn't that work?